File: ZCMS_FrontEnd/src/app/service/content-model.service.ts
import { Injectable } from '@angular/core';
import {ContentModel} from "../model/ContentModel";
import {HttpClient, HttpParams} from "@angular/common/http";
import {Observable} from "rxjs";
import {UrlConstant} from "../const/UrlConstant";
import {SystemConstant} from "../const/SystemConstant";
import {AuthService} from "./auth.service";

@Injectable({
  providedIn: 'root'
})
export class ContentModelService {

  private defaultUrl: string = SystemConstant.BASE_PATH + '/content-models';

  constructor(private httpClient: HttpClient, private authService: AuthService) { }

  getAllContentModelByCurrentUser(): Observable<ContentModel[]>{
    const userId: string = this.authService.getUserFromLocalStorage().id;
    const httpParam = new HttpParams().set("userId", userId);
    return this.httpClient.get<ContentModel[]>(this.defaultUrl, {params: httpParam});
  }

  createNewContentModel(contentModel: ContentModel): Observable<ContentModel>{
    return this.httpClient.post<ContentModel>(this.defaultUrl, contentModel);
  }

  updateContentModel(contentModel: ContentModel): Observable<ContentModel>{
    return this.httpClient.put<ContentModel>(this.defaultUrl, contentModel);
  }

  deleteContentModelById(id: string){
    const url = this.defaultUrl + '/' + id;
    return this.httpClient.delete(url);
  }
}

File: ZCMS_Backend/src/main/java/com/devz/zcms/service/ContentModelService.java
package com.devz.zcms.service;

import com.devz.zcms.entity.ContentModel;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ContentModelService {
    Flux<ContentModel> getAllContentModels();
    Flux<ContentModel> findAllContentModelsById(String userId);
    Mono<ContentModel> createNewContentModel(ContentModel contentModel);
    Mono<ContentModel> updateContentModel(ContentModel contentModel);
    Mono<Void> deleteContentModelById(String id);
}
